### PR TITLE
Add more SQL template fields renderers

### PR DESCRIPTION
**Pull request by josh-fell**
Related: #11177

This PR adds the `template_fields_renderers` attr to many operators as well as adds a few more lexers for syntax highlighting in the UI:

- `mysql`
- `postgresql`
- `tsql` (for Transact-SQL used in SQL Server)

> Note: The `ddl_statements` arg in the Google Cloud Spanner operators would be a list and does not render correctly in the UI currently. The list of SQL renders as a typical Python list rather than being separated into individual SQL statements and rendered like in the Task Instance Details page. However, #21227 _may_ fix this.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/www/utils.py [hunk 1]
```diff
@@ -414,21 +414,24 @@ def get_attr_renderer():
     return {
         'bash': lambda x: render(x, lexers.BashLexer),
         'bash_command': lambda x: render(x, lexers.BashLexer),
-        'hql': lambda x: render(x, lexers.SqlLexer),
```
Comment (josh-fell) [inline airflow/www/utils.py:417]: Doing some alphabetizing here along with adding 3 lexers.

File: docs/spelling_wordlist.txt [hunk 2]
```diff
@@ -1259,6 +1261,7 @@ sharded
 shellcheck
 shellcmd
 shm
+shouldn
```
Comment (uranusjr) [inline docs/spelling_wordlist.txt:1269]: Where is this used?

**Review by uranusjr — APPROVED** [review]

File: docs/spelling_wordlist.txt (hunk 2 above)
Comment (josh-fell) [reply docs/spelling_wordlist.txt:1269]: The spellchecker didn't like the contraction "shouldn't", and "aren't" either, in a few docstrings.
image

Comment (uranusjr) [reply docs/spelling_wordlist.txt:1269]: Hmm, it’s strange this is showing up here; this PR is not even changing baseoperator.py.

**Comment by josh-fell** [comment]
Rebased to resolve conflicts.

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
@josh-fell, can you rebase on main? The mssql failures should be fixed now.

**Pull request merged**